rollout/sync.go

```diff
@@ -578,6 +578,19 @@ func isIndefiniteStep(r *v1alpha1.Rollout) bool {
 	return false
 }
 
+// isWaitingForReplicaSetScaleDown returns whether or not the rollout still has other replica sets with a scale down deadline annotation
+func isWaitingForReplicaSetScaleDown(r *v1alpha1.Rollout, newRS, stableRS *appsv1.ReplicaSet, allRSs []*appsv1.ReplicaSet) bool {
```

Comment:
During this check, do you know if the `r.status.currentPodHash == r.status.stableRS` if so that means that we are finished or are not in the middle of the rollout wondering if it would make sense to add this check?

Reply:
this would then become isWaitingForReplicaSetScaleDownAndRolloutCompleted

Reply:
I can check if that's the case

Reply:
I did a bit of checking around deployment behavior and what you are doing here seems logical. Deployments can throw progressDeadline not during a "deployment" on a few cases and so I don't think we would want to add a check that limits adding a progressDeadline if we are not in the middle of a rollout.

Reply:
This was an interesting thread around k8s deployment behavior, https://github.com/kubernetes/kubernetes/issues/106054

Reply:
@zachaller thanks for looking at this.  Looking forward to `v1.7` :)
let me know if I can do anything else.